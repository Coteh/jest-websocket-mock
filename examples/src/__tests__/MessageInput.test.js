import React from "react";
import renderer from "react-test-renderer";
import makeStore from "../store";
import { actions } from "../store/reducer";
import MessageInput from "../MessageInput";

describe("The MessageInput component", () => {
  it("renders a form", () => {
    const store = makeStore();
    const component = renderer.create(<MessageInput store={store} />);
    expect(component).toMatchSnapshot();
  });

  it("sends the message when submitting the form", () => {
    const store = makeStore();
    jest.spyOn(store, "dispatch").mockImplementation(jest.fn());
    const component = renderer.create(<MessageInput store={store} />);
    component.root
      .findByType("input")
      .props.onChange({ target: { value: "Hello there" } });
    component.root.findByType("form").props.onSubmit(new Event("submit"));
    expect(store.dispatch).toHaveBeenCalledWith(actions.send("Hello there"));
  });
});
