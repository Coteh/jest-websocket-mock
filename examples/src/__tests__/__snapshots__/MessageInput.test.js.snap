// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MessageInput component renders a form 1`] = `
<form
  className="MessageForm"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="MessageInput"
    onChange={[Function]}
    value=""
  />
</form>
`;
